tools/cli/commands/changelog.js

```diff
@@ -50,19 +51,21 @@ export function changelogDefine( yargs ) {
 export async function changeloggerCli( argv ) {
 	// @todo Add validation of changelogger commands? See projects/packages/changelogger/README.md
 	// @todo refactor? .github/files/require-change-file-for-touched-projects.php to a common function that we could use here. Would allow us to run a "jetpack changelog add" without a project to walk us through all of them?
-	validateArgs( argv );
+	validateCmd( argv );
+
 	argv = normalizeProject( argv );
 	argv = await promptForProject( argv );
 	const projDir = path.resolve( `projects/${ argv.project }` );
-	const process = child_process.spawnSync( `vendor/bin/changelogger ${ argv.cmd }`, [ '' ], {
+	validatePath( argv, projDir );
+
+	const data = child_process.spawnSync( `vendor/bin/changelogger`, [ `${ argv.cmd }` ], {
 		cwd: projDir,
 		stdio: 'inherit',
-		shell: true,
 	} );
 
 	// Node.js exit code status 0 === success
-	if ( process.status !== 0 ) {
-		console.error( chalk.red( 'Something went wrong! Check your file path?' ), process.error );
+	if ( data.status !== 0 ) {
+		console.error( chalk.red( `Something went wrong! Exit Code: ${ data.status }` ), data.error );
```

Comment:
That's not what I meant about the exit status.

First, we need to break something in a way that will make `changelogger add` fail. This will do it, for example, by pointing the directory for change files to a file that already exists.
```
(cd projects/packages/codesniffer/ && composer config extra.changelogger.changes-dir CHANGELOG.md)
```

Then we run the CLI, and check the exit status:
```
$ jetpack changelog add packages/codesniffer
Could not create directory /usr/local/src/automattic/jetpack/projects/packages/codesniffer/CHANGELOG.md: mkdir(): File exists
Something went wrong! Exit Code: 1 undefined
$ echo $?
0
```
That needs to be 1 at the end, not 0. You can do that by adding `process.exit( data.status );` after this line. I don't know if there's a better way.

Reply:
Nice! Thanks for the explanation. Latest commit should do that.

Reply:
I haven't tested to see if the latest commit did what Brad suggested, but another way of doing it is to throw an error. Within the CLI, any uncaught throw will end up here https://github.com/Automattic/jetpack/blob/master/tools/cli/bin/jetpack#L15

With each command, we could have a catch that would grab any error from the command to give it a more contextual error.

Reply:
It's still returning exit code 0. If you follow the steps above:
```
(cd projects/packages/codesniffer/ && composer config extra.changelogger.changes-dir CHANGELOG.md)
jetpack changelog add packages/codesniffer
echo $?
```

We want to see a 1 there, but it's still 0. Throwing and catching, or to Brad's point, a process.exit would be good.

Reply:
A commit wasn't pushed. All better.